analyzers/ImpactParameter.py

```diff
@@ -1,48 +1,233 @@
 from heppy.framework.analyzer import Analyzer
-from ROOT import TVector3
-#from heppy.framework.event import Event
+from ROOT import TFile, TH1F
+from ROOT import TVector3, TLorentzVector
 from heppy.papas.path import Helix
-#import collections
+import math
+from heppy.utils.deltar import deltaR
 
 class ImpactParameter(Analyzer):
-
+    '''This analyzer puts an impact parameter for every charged particle
+    as an attribute of its path : particle.path.IP
+    The significance is calculated, the calculus are a first order approximation,
+    thus this may not be correct for large impact parameters (more than 3 mm)
+    It also computes W, the likelihood value for the jet to contain beauty,
+    using a likelihood ratio method for each particle in the jet.
+    Computes TCHE and TCHP for the jet for b-tagging'''
         
     def process(self, event):
         assumed_vertex = TVector3(0, 0, 0)
         jets = getattr(event, self.cfg_ana.jets)
+        enable_matter_scattering = self.cfg_ana.enable_matter_scattering
```

Comment:
is it really needed to declare a local variable? if possible just use self.cfg_ana.enable_matter_scattering instead